.github/workflows/post-pull-request-checks-automation.yml

```diff
@@ -40,13 +40,32 @@ on:
         type: number
         default: 1
         description: Configuration for the timeout-minutes parameter of the workflow.
+      run-only-latest-workflow:
+        type: string
+        description: Run only if the current workflow is the most recent workflow.
+        default: true
 
 jobs:
   post-pull-request-checks-automation:
     runs-on: ubuntu-22.04
     timeout-minutes: ${{ inputs.timeout-minutes }}
     steps:
+      - name: Checkout
```

Comment:
As far as I have tested things this is a must here because without this I cannot access data in the composite action that I need for it to work properly. As soon as I added checkout everything started working.

Reply:
Hmm, this shouldn't be necessary unless you want to run actions from the OSOCE repo, which is not the case. And GHA will automatically download the LGHA repo's content even without a checkout step (but then again, this one here doesn't do that), see:

![image](https://github.com/Lombiq/GitHub-Actions/assets/1976647/da4356d9-7c31-4124-aaba-252c021c1640)

Also, there's no `secrets.CHECKOUT_TOKEN` here.

Please show what exactly happens without this checkout.

Reply:
@Piedone 
![image](https://github.com/Lombiq/GitHub-Actions/assets/92299130/abfb3838-1c92-4377-b30e-18e5c853517d)
This happens when there is no Checkout. GH commands basically don't work. This line is perfect is there is checkout.

Reply:
This is the result with checkout:
![image](https://github.com/Lombiq/GitHub-Actions/assets/92299130/c8169df7-87d6-471c-ae81-ef1f1717ff71)
You can see at `line #22` that the echo did bring back the variable's value and it didn't error out.

Reply:
IF you ask me
![image](https://github.com/Lombiq/GitHub-Actions/assets/92299130/741a9b87-90a2-4932-bd66-9247deb71905)
The `failed to determine base repo:  failed to run git` part is the one to blame. It seems that for some reason we cannot access the GH CLI in a `composite actions` unless we are checking out in the REPO. 

By the way you mentioned `Having to checkout the OSOCE repo to run this workflow that does nothing with the OSOCE repo's content doesn't make sense.` 
Which is not 100% correct as the GH run list CLI command needs to access the workflows that are running in the `OSOCE` repository so if I am not mistaken we will need a checkout here.

Reply:
So ultimately `github context` is passed through and can be accessed from a `composite action` without checking out but the `GH CLI` needs the repository to function.

Reply:
Perhaps you need to pass `GITHUB_TOKEN` to the action from the workflow? Actions don't automatically have access to the exact same context as workflows, and thus they won't have this available in the environment either. And [the docs](https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows) tells this too:

> GitHub CLI is preinstalled on all GitHub-hosted runners. For each step that uses GitHub CLI, **you must set an environment variable called GITHUB_TOKEN** to a token with the required scopes.

Reply:
![image](https://github.com/Lombiq/GitHub-Actions/assets/92299130/aceb6d50-a172-45a1-961f-7854054c3281)
This seems to be the solution here.
So basically unless we are checking out we need to specify in the `gh run list` command in which repository do we want to list the workflow. So `--repo ${{ github.repository }}` is a must when using `GH CLI`

Reply:
@Piedone By the way there is a new flaw.
Which can be omitted with a bit more accurate querying of the runs.
So basically if you now put the `done-jira-issue-if-checks-succeed` to the PR and you expect this workflow to work. It will but it won't as the done-jira tag will trigger a skipped windows build which will eventually be a more recent workflow.
I am sure you want to get something be done about that.

Reply:
So this breaks how we work. This should be addressed before we can use this, yes.

Reply:
Yes this is an option that you said but in some wierd scenarios the following could happen:
1)
Is this the latest `Build and Test ubuntu latest` ? If yes and it passed put the issue to done
BUT
What if there is a pending `Build and Test Windows` that will fail for some reason. We put the issue to done but there is a failing build.

I investigated the switches and there really isn't a good solution for it.
We could try to play with the `--status` switch but that isn't easy either as the Windows build will be mostly either 
`skipped|queued|in_progress` and the `status` flag will accept only 1.

It would be the best to have something like `gh run list --allbutskipped` but that is basically a `!skipped` which doesn't exist. 

Either way if we really want to count in the windows workflow it will be a bit messy.

Reply:
We don't need to care about Windows builds for now. It's extremely rare that you run it together with the Ubuntu ones, since the PR author most of time won't run them, and the PR reviewer will only run them at the end (at which point the latest Ubuntu build for the most recent commit should already be finished).

So, the logic here can be what I've written, i.e. the most recent workflow of the same name, for the current branch.